### PR TITLE
Mpirnovar/port acceptance tests

### PR DESCRIPTION
**Pull request by Mat001**
## Summary
@mikecdavis  @juancarlostong 

It's not pretty. Lots to be cleaned out. But it works locally.
Run from agent's root. Or see Travis build.
`pytest -v tests/acceptance/test_acceptance/ --host http://localhost:8080`

Notes:

TWO .gitignore files now! - many files in my gitignore should probably be added to the root gitignore
@juancarlostong Can you pls check .travis.yaml change.

### CODE REVIEW

File: .idea/agent.iml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (Mat001) [inline .idea/agent.iml]: .idea stuff should not be added to the repo. I'm suggesting to update the root .gitignore with "gitignore" things that acceptance tests introduce.

File: .idea/.gitignore [hunk 2]
```diff
@@ -0,0 +1,2 @@
+# Default ignored files
```
Comment (mikecdavis) [inline .idea/.gitignore]: Please remove all of the `.idea/` files from git.

File: .travis.yml [hunk 3]
```diff
@@ -81,6 +81,21 @@ jobs:
       after_success:
         - ( [ ${TRAVIS_EVENT_TYPE} = "push" ] && [ ! -z ${TRAVIS_TAG} ] ) && scripts/upload_artifacts.sh
 
+    - stage: Test
```
Comment (mikecdavis) [inline .travis.yml]: Should probably be it's own stage, like `"Integration"`

File: tests/acceptance/.gitignore [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+/site
+
+# mypy
+.mypy_cache/
+.dmypy.json
+dmypy.json
+
+# Pyre type checker
+.pyre/
+# Local .terraform directories
```
Comment (mikecdavis) [inline tests/acceptance/.gitignore]: It looks like everything after this line is in addition to the base python gitignore from [here](https://github.com/github/gitignore/blob/master/Python.gitignore). I don't think it's necessary and can be removed.

File: tests/acceptance/test_acceptance/__init__.py [hunk 5]
```diff
@@ -0,0 +1 @@
+# __init__ is empty
```
Comment (mikecdavis) [inline tests/acceptance/test_acceptance/__init__.py]: Nit, this module name feels redundant. Should we name it `tests/acceptance/test`? Also is this string needed?

File: tests/acceptance/test_acceptance/test_config.py [hunk 6]
```diff
@@ -0,0 +1,66 @@
+import os
+
+from tests.acceptance.helpers import ENDPOINT_CONFIG
+
+BASE_URL = os.getenv('host')
+
+# TODO use json vs dict
+# -> more portable
+
+expected_config = {'revision': '111', 'experimentsMap': {
```
Comment (mikecdavis) [inline tests/acceptance/test_acceptance/test_config.py]: Another good place to defer to a `.json` file as opposed to a python dict.

File: tests/acceptance/README.md [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+
+
+
+
+
+
+
+
+
+
```
Comment (mikecdavis) [inline tests/acceptance/README.md]: Please remove all of these superfluous lines.

File: tests/acceptance/helpers.py [hunk 8]
```diff
@@ -0,0 +1,147 @@
...
+import os
+import string
+import time
+from random import randint, choice
+
+import requests
+
+ENDPOINT_ACTIVATE = '/v1/activate'
+ENDPOINT_CONFIG = '/v1/config'
+ENDPOINT_NOTIFICATIONS = '/notifications/event-stream'
```
Comment (mikecdavis) [inline tests/acceptance/helpers.py]: Should be `/v1/notifications/event-stream` now.

**Pull request merged**